tests/app/test_create_app.py

```diff
@@ -0,0 +1,76 @@
+"""Verifies that instance_links are being retrieved properly from LINKS. Verifies that app_data.json.j2
+contains the instance link information"""
+from unittest import mock
+import requests_mock
+import json
+import foremast.app.create_app
+from foremast.app import SpinnakerApp
+
+
+@mock.patch('foremast.app.create_app.LINKS', new={"test1": "https://test1.com", "test2": "https://test2.com"})
+def test_instance_links():
+    """Checks if instance_links is being populated properly"""
+    pipeline_config = {
+        "instance_links": {
+            "test3": "https://test2.com",
+            "test4": "https://test4.com",
+        }
+    }
+    spinnaker_app = SpinnakerApp(pipeline_config=pipeline_config)
+    instance_links = spinnaker_app.retrieve_instance_links()
+    assert instance_links == {"test1": "https://test1.com"}, "Instance Links are not being retrieved properly"
```

Comment:
I am not sure I understand this test. I would expected instance_links to equal `test3` and `test4` right? Why are you asserting that they equal `test`?

Reply:
this is because this first test is simply testing the retrieve_instance_links method which returns a dictionary containing all items not in pipeline_config.

Reply:
The returned value is then added into pipeline_configs

Reply:
If that is the case, shouldnt it return `test2` as well?

Reply:
Nope so this method simply returns all the items in LINKS and not in pipeline_configs. This dictionary is then added into pipeline_configs to provide a dictionary that contains only unique items

Reply:
But you set LINKS to `'foremast.app.create_app.LINKS', new={"test1": "https://test1.com", "test2": "https://test2.com"}`. So why is `test1` and `test2` not returned?

Reply:
because "https://test2.com" is in pipeline_configs['instance_links']. I test to see if two values are the same in either dictionary.

Reply:
Where are you seeing `test2` being in `pipeline_configs['instance_links']`? I see test3 and test4.

Reply:
OHHH the url. You should make this more clear. Part of a unit test is to help understand how the code works

Reply:
and what is the string `"Instance Links are not being retrieved properly"` at the end of the assert doing?

Reply:
Sorry Ill include better comments for the unit tests. The string afterwards explains why the two values are not equal. Essentially why the unit test did not pass

Reply:
okay looks like i fixed that